x/ecocredit/server/msg_server.go

```diff
@@ -373,35 +378,6 @@ func (s serverImpl) Cancel(goCtx context.Context, req *ecocredit.MsgCancel) (*ec
 	return &ecocredit.MsgCancelResponse{}, nil
 }
 
-func (s serverImpl) SetPrecision(goCtx context.Context, req *ecocredit.MsgSetPrecision) (*ecocredit.MsgSetPrecisionResponse, error) {
-	ctx := types.UnwrapSDKContext(goCtx)
-	var batchInfo ecocredit.BatchInfo
-	err := s.batchInfoTable.GetOne(ctx, orm.RowID(req.BatchDenom), &batchInfo)
-	if err != nil {
-		return nil, err
-	}
-	if req.Issuer != batchInfo.Issuer {
-		return nil, sdkerrors.ErrUnauthorized
-	}
-	store := ctx.KVStore(s.storeKey)
-	key := MaxDecimalPlacesKey(batchDenomT(req.BatchDenom))
```

Comment:
We also need to look in `CreateBatch`, `Retire`, and `Cancel` at the places where `MaxDecimalPlacesKey` is being used and replace that with `CreditType` based precision checks.

Reply:
And then we can remove all references to `MaxDecimalPlacesKey`

Reply:
how strictly should we enforce the precision - should it be a hard conditional like:

`tradable decimal places must == credit type precision`

or

`tradable decimal places must be <= credit type precision`?

Reply:
Definitely `<=`! We don't want to force people to write `1.000000` when they want to send 1 credit!